sentry/src/main/java/io/sentry/logger/LoggerBatchProcessor.java

```diff
@@ -0,0 +1,101 @@
+package io.sentry.logger;
+
+import io.sentry.ISentryClient;
+import io.sentry.ISentryLifecycleToken;
+import io.sentry.SentryLogEvent;
+import io.sentry.SentryLogEvents;
+import io.sentry.SentryOptions;
+import io.sentry.util.AutoClosableReentrantLock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Future;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class LoggerBatchProcessor implements ILoggerBatchProcessor {
+
+  public static final int FLUSH_AFTER_MS = 5000;
+  public static final int MAX_BATCH_SIZE = 100;
+
+  private final @NotNull SentryOptions options;
+  private final @NotNull ISentryClient client;
+  private final @NotNull Queue<SentryLogEvent> queue;
+  private volatile @Nullable Future<?> scheduledFlush;
+  private static final @NotNull AutoClosableReentrantLock scheduleLock =
+      new AutoClosableReentrantLock();
+
+  public LoggerBatchProcessor(
+      final @NotNull SentryOptions options, final @NotNull ISentryClient client) {
+    this.options = options;
+    this.client = client;
+    this.queue = new ConcurrentLinkedQueue<>();
+  }
+
+  @Override
+  public void add(final @NotNull SentryLogEvent logEvent) {
+    queue.offer(logEvent);
+    maybeSchedule(false, false);
```

Comment:
This means we're locking on every log event. I'll try to optimize this. Ideas:

1. Have a bool `hasScheduled` or similar which we check first and avoid locking if already true
2. Check the futures state first (not sure about performance of that)
3. Just send a signal and have a background thread pick it up by periodically polling for it